tests/test_logger.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+import pytest
+
+from app.internal.logger_customizer import LoggerCustomizer, LoggerConfigError
+
+
+class TestLogger:
+    @staticmethod
+    def test_log_debug(caplog, client_with_logger):
+        with caplog.at_level(logging.DEBUG):
+            client_with_logger.logger.debug('Is it debugging now?')
+            assert 'Is it debugging now?' in caplog.text
+
+    @staticmethod
+    def test_log_info(caplog, client_with_logger):
+        with caplog.at_level(logging.INFO):
+            client_with_logger.logger.info('App started')
+            assert 'App started' in caplog.text
+
+    @staticmethod
+    def test_log_error(caplog, client_with_logger):
+        with caplog.at_level(logging.ERROR):
+            client_with_logger.logger.error('Something bad happened!')
+            assert 'Something bad happened!' in caplog.text
+
+    @staticmethod
+    def test_log_critical(caplog, client_with_logger):
+        with caplog.at_level(logging.CRITICAL):
+            client_with_logger.logger.critical("WE'RE DOOMED!")
+            assert "WE'RE DOOMED!" in caplog.text
```

Comment:
You could save some space by changing this to something like this:

```
LOGGER_TESTS = [
    (logging.DEBUG, 'Is it debugging now?'),
    (logging.INFO, 'App started'),
]

@staticmethod
@pytest.mark.parametrize("logger_level, message", LOGGER_TESTS)
def test_log_debug(caplog, client_with_logger, logger_level, message):
    with caplog.at_level(logger_level):
        client_with_logger.logger.debug(message)
        assert message in caplog.text
```

Reply:
Totally been thinking about it, you missed the line that made me not do it though:
client_with_logger.logger.**debug**('Is it debugging now?')
I couldn't figure out how to parametrize a fixture's function, tried to do something like that:

```
@staticmethod
    @pytest.mark.parametrize("logger_func, logger_level, message", LOGGER_TESTS)
    def test_log_debug(caplog, logger_instance, logger_func, logger_level, message):
        LOGGER_TESTS = [
            (logger_instance.debug, logging.DEBUG, 'Is it debugging now?'),
            (logger_instance.info, logging.INFO, 'App started'),
            (logger_instance.error, logging.ERROR, 'Something bad happened!'),
            (logger_instance.critical, logging.CRITICAL, "WE'RE DOOMED!")
        ]
        with caplog.at_level(logger_level):
            logger_func(message)
            assert message in caplog.text
```

but then, the fixture isn't received by the test function.
If I put the parameters outside the inner fixture functions aren't known.
So I started thinking of a wrapper function but it seems like an overkill to me.